scai-gen/fileio/common.go

```diff
@@ -0,0 +1,17 @@
+package fileio
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func HasJSONExt(filename string) bool {
+	return strings.HasSuffix(filename, ".json")
+}
+
+func CreateOutDir(filename string) error {
+	outDir := filepath.Dir(filename)
+
+	return os.MkdirAll(outDir, 0644)
```

Comment:
Does MkdirAll return no error if the dir / part of it exists? I think it must because it might have to create directories even when others do. :thinking:

Reply:
> MkdirAll creates a directory named path, along with any necessary parents, and returns nil, or else returns an error. The permission bits perm (before umask) are used for all directories that MkdirAll creates. If path is already a directory, MkdirAll does nothing and returns nil.

The docs seem to imply this is fine.

Reply:
Yep! It essentially seems to mimic the behavior of `mkdir -p`.